sheer/indexer.py

```diff
@@ -149,8 +149,21 @@ def index_location(args, config):
                                 doc_type=processor.name,
                                 body={processor.name: processor.mapping(default_mapping)})
 
+        # Get the document iterator from the processor.
+        # This could raise a value error raised by json.loads() if there is a
+        # connection error, server error, or other error acquiring the data.
+        # This is seperate from the for loop so that we dont create/update some
+        # of the documents then error.
+        try:
+            document_iterator = enumerate(processor.documents())
+        except ValueError:
+            # If there's an error, log it, and then move on to the next content
+            # processor.
+            sys.stderr.write("error reading documents for %s" % processor.name)
+            continue
+
         i = -1
-        for i, document in enumerate(processor.documents()):
+        for i, document in document_iterator:
```

Comment:
1. Yea I hear you, wrapping the whole loop doesn't seem right.  I like the way breaking out the function and wrapping that sounds.  
2. When this error does happen, it usually happens before any documents are indexed.  In the case that it would error partway through, I agree that it would be cleaner to not have already indexed some of the documents, but we'd need to do a full reindex of that processor anyway, right?